### PR TITLE
[Merged by Bors] - chore(SpecialFunctions): re-instate many simp lemmas

### PR DESCRIPTION
**Pull request by grunweg**
Remove some now-addressed porting notes; re-instate a few more lemmas which were formerly marked as simp.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by j-loreaux** [comment]
@mattrobball what does your keying tool say about these `simp` lemmas? If you point me to it, I can check myself.

**Comment by mattrobball** [comment]
`#discr_tree_simp_key my_lemma`

File: Mathlib/MeasureTheory/Integral/IntervalIntegral.lean [hunk 1]
```diff
@@ -620,12 +620,7 @@ section Comp
 
 variable {a b c d : ℝ} (f : ℝ → E)
 
-/-!
-Porting note: some `@[simp]` attributes in this section were removed to make the `simpNF` linter
-happy. TODO: find out if these lemmas are actually good or bad `simp` lemmas.
--/
-
-@[simp] -- Porting note (#10618): was @[simp]
+@[simp]
```
Comment (mattrobball) [inline Mathlib/MeasureTheory/Integral/IntervalIntegral.lean:623]: `@intervalIntegral _ _ _ <other> _ _ (@volume ℝ _)`

File: Mathlib/MeasureTheory/Integral/IntervalIntegral.lean [hunk 2]
```diff
@@ -638,32 +633,32 @@ theorem integral_comp_mul_right (hc : c ≠ 0) :
       Measure.restrict_congr_set (α := ℝ) (μ := volume) Ico_ae_eq_Ioc]
   · simp [h, mul_div_cancel_right₀, hc, abs_of_pos]
 
-@[simp] -- Porting note (#10618): was @[simp]
+@[simp]
```
Comment (mattrobball) [inline Mathlib/MeasureTheory/Integral/IntervalIntegral.lean:636]: `@HSMul.hSMul ℝ _ _ _ _ (@intervalIntegral _ _ _ <other> _ _ (@volume ℝ _))`

File: Mathlib/MeasureTheory/Integral/IntervalIntegral.lean [hunk 3]
```diff
@@ -638,32 +633,32 @@ theorem integral_comp_mul_right (hc : c ≠ 0) :
...
 theorem smul_integral_comp_mul_right (c) :
     (c • ∫ x in a..b, f (x * c)) = ∫ x in a * c..b * c, f x := by
   by_cases hc : c = 0 <;> simp [hc, integral_comp_mul_right]
 
-@[simp] -- Porting note (#10618): was @[simp]
+@[simp]
```
Comment (mattrobball) [inline Mathlib/MeasureTheory/Integral/IntervalIntegral.lean:641]: `@intervalIntegral _ _ _ <other> _ _ (@volume ℝ _)`

File: Mathlib/MeasureTheory/Integral/IntervalIntegral.lean [hunk 4]
```diff
@@ -638,32 +633,32 @@ theorem integral_comp_mul_right (hc : c ≠ 0) :
...
 theorem integral_comp_mul_left (hc : c ≠ 0) :
     (∫ x in a..b, f (c * x)) = c⁻¹ • ∫ x in c * a..c * b, f x := by
   simpa only [mul_comm c] using integral_comp_mul_right f hc
 
-@[simp] -- Porting note (#10618): was @[simp]
+@[simp]
```
Comment (mattrobball) [inline Mathlib/MeasureTheory/Integral/IntervalIntegral.lean:646]: `@HSMul.hSMul ℝ _ _ _ _ (@intervalIntegral _ _ _ <other> _ _ (@volume ℝ _))`

**Review by mattrobball — COMMENTED** [review]
I went through a couple and the keys aren't great for the context.

**Comment by mattrobball** [comment]
!bench

**Comment by j-loreaux** [comment]
If we wanted to, we could still scope them to `intervalIntegral`, although in that case, it would be nice if there were a comment in the docstring for each lemma.

**Comment by mattrobball** [comment]
Since `intervalIntegral` is the head symbol, the discrimination tree already effectively does the scoping. The main issue is that it doesn't see anything beyond that head symbol for the most part. I think these are ok for now because the domain is pretty small. It will probably be fine indefinitely.

**Comment by grunweg** [comment]
I do not have a strong opinion about the newly marked simp lemmas: I am happy to also just remove the porting note.

**Comment by mattrobball** [comment]
I think just merging is fine.

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
bors r-

bors d+

**Comment by bryangingechen** [comment]
One of the runners ran out of space (and I've temporarily removed it from the pool), so re-running this should work. I'm not sure if there's another reason this was given `bors r-` though.

**Pull request closed without merging**